src/Microsoft.DotNet.Build.Tasks.Packaging/src/PackageFiles/Packaging.targets

```diff
@@ -493,19 +493,14 @@
         <PackageVersion>$(PackageVersion)</PackageVersion>
       </PackageFile>
 
-      <!-- Include Sources. Deduplicate so the nuspec doesn't contain multiple entries for each source file. -->
-      <PackageSources Include="@(File)"
-                      KeepMetadata="TargetPath;IsSourceCodeFile"
-                      KeepDuplicates="false"
-                      Condition="'%(File.IsSourceCodeFile)'=='true'" />
-      <!-- Add a placeholder source file if there are symbols but no sources. -->
-      <PackageSymbolFiles Include="@(File)" Condition="'%(File.IsSymbolFile)'=='true'" />
+      <!-- Add a placeholder source file if there are symbols but no sources, so that MyGet will treat symbol packages as valid -->
       <PackageSources Include="$(PlaceholderFile)" Condition="'@(PackageSymbolFiles)'!='' AND '@(PackageSources)'==''">
```

Comment:
I just realized that since this will only add the placeholder with `'@(PackageSymbolFiles)'!=''`, we probably also need to keep `<PackageSymbolFiles Include="@(File)" Condition="'%(File.IsSymbolFile)'=='true'" />`.

Have you tried this out in a build? What we're looking for is the nupkg having no `src/_._` placeholder, and the symbols.nupkg having some `**/*.pdb` file(s) and `src/_._`.

Reply:
My last build of this looked right for packages, but there is no `src/_._` in the symbol packages. I'll try adding that line back in & rebuilding.

Reply:
@dagood there is now a placeholder in both normal & symbol packages. Should I undo https://github.com/dotnet/buildtools/pull/2090/commits/47007205f0abddd29e96dd4d1f7bdc7cd90a9010?

Reply:
I don't see how that would fix it, but if it does, go for it. I'd have to do some stuff myself with binlogs, diag logs, `/pp`, etc. to figure out exactly what to do.

Reply:
Can also use the LaunchDebugger buildtools task to figure out why NuGetPack is/isn't filtering certain things out.

Reply:
As far as I can tell from the binlog/playing around with the targets, we can either have the placeholder in both places, or in neither (without making more significant changes). Is there a real harm to having a placeholder file in the normal package?

Reply:
The harm is potential flakiness for transport packages (since they include PDBs) on MyGet. This is what's happening in https://github.com/dotnet/coreclr/issues/13798.

Without this PR, are you also seeing that behavior? It doesn't seem right to give up and regress now. 😕

Reply:
Without this PR, both normal packages & symbol packages contain the full set of sources. I haven't yet been able to make a change to ones `src` directory without making the same change to the other's. I can keep poking around for a way though, if it will be a problem to have a placeholder in the non-symbol packages.

Reply:
Note that the transport packages are also ["special"](https://github.com/dotnet/corefx/blob/0b0f63d47951773cc46393312d80b40176d3a413/pkg/Microsoft.Private.CoreFx.NETCoreApp/Microsoft.Private.CoreFx.NETCoreApp.pkgproj#L16) so they may be a bit confusing to look at.  For those ones I expect symbols and source in both sets of packages.  Make sure you look at a regular package as well.

Reply:
I've been building CoreFx so I haven't seen a transport package yet. I'll try building CoreClr.

Reply:
Ah, I think you're looking at transport packages (rightly) and I'm thinking more about e.g. `System.IO.Ports` which doesn't have sources (or symbols) https://dotnetfeed.blob.core.windows.net/dotnet-core/flatcontainer/system.io.ports/4.6.0-preview1-26712-04/system.io.ports.4.6.0-preview1-26712-04.nupkg

I think this is happening because of `NuGetPack`'s `IncludeSymbolsInPackage`, which I think is being used to make the transport package. It's what filters the items down and excludes the placeholder, so I think it makes sense to change the task code itself (and probably delete this item).

Reply:
@wtgodbe we call those "private" packages transport packages, so you have seen them.

Reply:
(Missed last two comments.) CoreFX's, transport packages are the `runtime.win10-x64.Microsoft.Private.CoreFx.UAP` ones--"transport package" just means it only delivers bits between repos and isn't published to the nuget.org gallery, kinda confusing that coreclr's have the `transport.` prefix and CoreFX's don't.

Reply:
Oh, to be clear, I've been looking at `Microsoft.Private.CoreFx.NETCore.App`.

If I look at `Microsoft.NETCore.Targets`, neither the normal package nor the symbol package have a placeholder file before or after my change

Reply:
That package has no dlls (so no src or symbols) only a runtime.json, use a library package.  Something System.*

Reply:
https://github.com/dotnet/buildtools/pull/1369 is a good place to understand the changes to NuGetPack to enable transport/private packages. Then I think something like this should be added (maybe not in the same if statement) that adds to `builder` if it needs to add a placeholder:

https://github.com/dotnet/buildtools/blob/185e36b4c1372d815af1b5a728249c0afdde90c0/src/Microsoft.DotNet.Build.Tasks.Packaging/src/NuGetPack.cs#L269-L273

---

Stepping back... we could just remove all src placeholders, and sweep through the `master` builds and make sure they all publish to the symbol-package-specific MyGet endpoint (if they publish to MyGet) so we don't need to care about the MyGet symbol package detection anymore. This is kind of risky because it's hard to tell immediately by browsing MyGet when things go wrong, but seems possible.

Package push endpoint: `https://dotnet.myget.org/F/dotnet-core/api/v2/package`
Symbol package push endpoint: `https://dotnet.myget.org/F/dotnet-core/symbols/api/v2/package`

Reply:
@ericstj I built System.Security.Cryptography.Xml & neither its package nor its symbol package had a `src` folder

@dagood so the 2 options are to modify NugetPack to inject a placeholder into symbol packages, or to not bother with a placeholder at all, and change how we publish to myget - but both options entail removing placeholders from this .targets file entirely?

Reply:
> so the 2 options are [...] both options entail removing placeholders from this .targets file entirely?

Yep. I can imagine other options that don't involve removing the placeholder MSBuild code (I imagine it requires more substantial changes to the `NuGetPack` logic to make sure our desired scenarios are covered) but those two options seem like the simplest to me.

Reply:
@dagood it looks like we already segregate symbol packages & normal packages into the symbol package/normal package myget endpoints (at least in master for the 3 core repos), so easiest would probably be to just merge this as-is (after removing this placeholder code)

Reply:
WCF doesn't: https://devdiv.visualstudio.com/DefaultCollection/DevDiv/_build/results?buildId=1857075. But apart from dotnet/standard (which doesn't even push symbol packages to MyGet), that's everyone using BuildTools, so I agree.

Reply:
That should be easy enough to change, they already have the package publish & symbol package publish divided into 2 steps. @StephenBonikowsky is it alright if I modify the symbol publish step to push to the myget symbols endpoint (`https://dotnet.myget.org/F/dotnet-core/symbols/api/v2/package`) instead of the package endpoint `https://dotnet.myget.org/F/dotnet-core/api/v2/package`)?

Reply:
We haven't been updating our Master branch since 2.1 released. I have a work item to base our Master on dependencies from 2.1 such as [buildtools/release/2.1](https://github.com/dotnet/buildtools/tree/release/2.1). So you can go ahead and make this change without updating our build definition.

Reply:
Sounds good. In that case, it sounds like all that needs to be done is to merge this PR. @dagood does that sound right to you?

Reply:
Yep. Thanks for the info @StephenBonikowsky!